raptor_dbw_can/src/DbwNode.cpp

```diff
@@ -848,9 +851,46 @@ void DbwNode::recvCAN(const can_msgs::msg::Frame::SharedPtr msg) {
                 }
             } break;
         }
+
+        // generates TireTemp messages and assign them to TireTempReport.
+        tire_temp_report_msg.front_left = DbwNode::generateTireTemp(fl_tire_temperature);
+        tire_temp_report_msg.front_right = DbwNode::generateTireTemp(fr_tire_temperature);
+        tire_temp_report_msg.rear_left = DbwNode::generateTireTemp(rl_tire_temperature);
+        tire_temp_report_msg.rear_right = DbwNode::generateTireTemp(rr_tire_temperature);
     }
 }
 
+// computes values of TireTemp message for a given array of tire temperatures.
+deep_orange_msgs::msg::TireTemp DbwNode::generateTireTemp(float32[] tire_temperature) {
+    deep_orange_msgs::msg::TireTemp tireTemp;
+    float32 sum = 0.0;
+    int count = 0;
+    float32[] temps;
+
+    for (int i = 0; i < 16; i++) {
```

Comment:
technically we _could_ optimize this such that the sort and averaging happens in one pass, but there are so few things to sort over that it should be fine. 

But, if you are up to it, would be a fun exercise to make this more efficient :)

Reply:
By one pass, I mean as you iterate over the whole tire temp array and keep track of valid temps, you could also sort the valid temps as you insert them into the new list.